internal/controllers/variable_source.go

```diff
@@ -38,5 +38,8 @@ func NewVariableSource(cl client.Client) variablesources.NestedVariableSource {
 		func(inputVariableSource input.VariableSource) (input.VariableSource, error) {
 			return variablesources.NewCRDUniquenessConstraintsVariableSource(inputVariableSource), nil
 		},
+		func(inputVariableSource input.VariableSource) (input.VariableSource, error) {
+			return variablesources.NewCatalogdVariableSource(cl), nil
+		},
```

Comment:
I think this might have to go higher in the slice. Each variable source in this slice nests the one above and some of the variable sources depend on variables received from the nested sources.

E.g. I believe `NewBundlesAndDepsVariableSource` depends on variables produced by catalog variable source.